src/looprestoration.rs

```diff
@@ -726,3 +726,223 @@ pub(crate) unsafe extern "C" fn selfguided_filter<BD: BitDepth>(
         }
     };
 }
+
+// TODO(randomPoison): Temporarily pub until init logic is deduplicated.
+pub(crate) unsafe extern "C" fn sgr_5x5_c<BD: BitDepth>(
+    mut p: *mut libc::c_void,
+    stride: ptrdiff_t,
+    left: *const libc::c_void,
+    mut lpf: *const libc::c_void,
+    w: libc::c_int,
+    h: libc::c_int,
+    params: *const LooprestorationParams,
+    edges: LrEdgeFlags,
+    bitdepth_max: libc::c_int,
+) {
+    sgr_5x5_rust(
+        p.cast(),
+        stride,
+        left.cast(),
+        lpf.cast(),
+        w,
+        h,
+        params,
+        edges,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+unsafe fn sgr_5x5_rust<BD: BitDepth>(
+    mut p: *mut BD::Pixel,
+    stride: ptrdiff_t,
+    left: *const [BD::Pixel; 4],
+    mut lpf: *const BD::Pixel,
```

Comment:
```suggestion
    lpf: *const BD::Pixel,
```

Same for the other `*_rust` `fn`s.